lib/rsolr/uri.rb

```diff
@@ -3,34 +3,25 @@
 module RSolr::Uri
   
   def create url
-    ::URI.parse url[-1] == ?/ ? url : "#{url}/"
+    ::URI.parse (url[-1] == '/' || URI.parse(url).query) ? url : "#{url}/"
   end
   
   # Returns a query string param pair as a string.
-  # Both key and value are escaped.
-  def build_param(k,v, escape = true)
+  # Both key and value are URI escaped, unless third param is false
+  # @param [boolean] escape false if no URI escaping is to be performed.  Default true.
+  def build_param(k, v, escape = true)
     escape ? 
-      "#{escape_query_value(k)}=#{escape_query_value(v)}" :
+      "#{URI.encode_www_form_component(k)}=#{URI.encode_www_form_component(v)}" :
       "#{k}=#{v}"
   end
 
-  # Return the bytesize of String; uses String#size under Ruby 1.8 and
-  # String#bytesize under 1.9.
-  if ''.respond_to?(:bytesize)
-    def bytesize(string)
-      string.bytesize
-    end
-  else
-    def bytesize(string)
-      string.size
-    end
-  end
-
   # Creates a Solr based query string.
   # Keys that have arrays values are set multiple times:
   #   params_to_solr(:q => 'query', :fq => ['a', 'b'])
   # is converted to:
   #   ?q=query&fq=a&fq=b
+  # @param [boolean] escape false if no URI escaping is to be performed.  Default true.
+  # @return [String] Solr query params as a String, suitable for use in a url
   def params_to_solr(params, escape = true)
```

Comment:
Huh - oddly, I just saw  #40 which states "Legacy Solr clients don't want escaped URLs, this pull request allows a client to be configured so that it will generate non-escaped URLs. "

Not sure I agree with the comment, but it's there.